pkg/asset/ignition/bootstrap/common.go

```diff
@@ -189,6 +192,12 @@ func (a *Common) generateConfig(dependencies asset.Parents, templateData *bootst
 
 	// Check for optional platform specific files/units
 	platform := installConfig.Config.Platform.Name()
+
+	// ARO is a special case of Azure
+	if installConfig.Config.Platform.Azure.IsARO() == true {
```

Comment:
This `IsARO` flag corresponds to an ARO-specific build of the installer, but my understanding of the goal is to get rid of the ARO fork so we would need some other way of configuring the installer to use this specific functionality, like an install config field. 

ARO uses dnsmasq for on-cluster DNS (cc @sadasu), but that is not generally supported in openshift. Are we trying to work toward [dnsmasq support in openshift](https://github.com/openshift/enhancements/pull/654)?

Reply:
@patrickdillon IMHO, it would be great to move in the direction of ARO supporting CoreDNS since that is the preferred solution for DNS within k8s. But I understand that the CoreDNS image is not available in ARO. (should we be fixing that 1st?)
So, we may have to limit dnsmaq use to within the ARO fork for now.

Reply:
To the extent that the DNS solution here is replicating the work that Sandhya is currently doing for multiple platforms (and it's not clear to me whether this is or not), I agree that we should aim to switch to that implementation when it is ready.

However, in any event we know that dnsmasq is also being used here (because it is built in to RHCOS) to resolve the registry IP in order to pull any container images, and that isn't going away.